src/Security/Authentication/OAuth/src/OAuthHandler.cs

```diff
@@ -211,7 +211,7 @@ protected virtual async Task<OAuthTokenResponse> ExchangeCodeAsync(OAuthCodeExch
         requestMessage.Content = requestContent;
         requestMessage.Version = Backchannel.DefaultRequestVersion;
         var response = await Backchannel.SendAsync(requestMessage, Context.RequestAborted);
-        var body = await response.Content.ReadAsStringAsync();
+        var body = await response.Content.ReadAsStringAsync(Context.RequestAborted);
```

Comment:
This doesn't have any effect, SendAsync doesn't return until the response is fully buffered.